services/ui-src/src/components/fields/DynamicField.tsx

```diff
@@ -163,10 +164,18 @@ export const DynamicField = ({ name, label, ...props }: Props) => {
                   ...newEntity,
                   plan_ilosUtilizationByPlan: filteredUtilizationByPlan,
                 };
+
+                // delete associated ILOS response
                 delete newEntity[
-                  `plan_ilosUtilizationByPlan-otherText_${selectedRecord.id}`
+                  `plan_ilosUtilizationByPlan_${selectedRecord.id}`
                 ];
+
+                // if there's no ILOS, clear checked response
+                if (selectedRecord.length < 1) {
```

Comment:
Since `selectedRecord` is of type `EntityShape`, what does it mean to look for a length of < 1? Does that affect other areas in this method, particularly all the `selectedRecord.id`s that are used?

Reply:
ah, that's my mistake. I'm referring to the wrong entity here, thanks for the catch! just updated it.